contrib/babelfishpg_tsql/src/hooks.c

```diff
@@ -2689,6 +2699,169 @@ pltsql_detect_numeric_overflow(int weight, int dscale, int first_block, int nume
+			/* To-Do Check if valid text month is given */
+			count1++;
+		else
+			return false;
+	}
+	if(count0 ==2 && count1 ==1)
+		return true;
+	return false;
+}
+
+Datum pltsql_time_in(const char *str, int32	typmod)
+{
+	TimeADT		result;
+	fsec_t		fsec;
+	struct pg_tm tt,
+			   *tm = &tt;
+	int			tz;
+	int			nf;
+	int			dterr;
+	char		workbuf[MAXDATELEN + 1];
+	char	   *field[MAXDATEFIELDS];
+	int			dtype;
+	int			ftype[MAXDATEFIELDS];
+	StringInfo	sf_1 ;
+	int status;
+
+	dterr = ParseDateTime(str, workbuf, sizeof(workbuf),
+						field, ftype, MAXDATEFIELDS, &nf);
+	if (dterr == 0)
+	{
+		if(nf >= 3)
+		{
+			if(containsInMonthFormat(ftype))
+			{
+				StringInfoData buf;
+				initStringInfo(&buf);
+				appendStringInfoString(&buf, field[0]);
+				for(int i=1;i<3;i++)
+				{
+					appendStringInfoString(&buf, "-");
+					appendStringInfoString(&buf, field[i]);
+
+				}
+				field[0]=buf.data;
+				ftype[0] = DTK_DATE;
+				nf = nf - 2;
+				for(int i = 1; i < nf; i++)
+				{
+					field[i] = field[i+2];
+					ftype[i] = ftype[i+2];
+				}
+			}
+		}
+
+		for(int i=0;i < nf; i++)
+		{
+			switch(ftype[i])
+			{
+				case DTK_NUMBER:
+					switch(strlen(field[i]))
+					{
+						case 1:
+						case 2:
+							/*
+							 * At this point the 1,2 digit number should be considered as
+							 * time only if there is an [AP]M after the number or else
+							 * should throw an error
+							 */
+							if(i == nf-1 || ftype[i+1] != DTK_STRING)
+							{
+								gen_func_time_in_error();
+							}
+							status = RE_compile_and_execute(cstring_to_text("^([AP]M)$"),
+															(char *) field[i+1], strlen(field[i+1]),
+															REG_ADVANCED | REG_ICASE, DEFAULT_COLLATION_OID,
+															0, NULL);
+							if(!status)
+								gen_func_time_in_error();
+
+							sf_1 = makeStringInfo();
+							appendStringInfo(sf_1, "%s%s", field[i], ":0:0");
+							field[i] = sf_1->data;
+							ftype[i] = DTK_TIME;
+							break;
+						/*
+						 * To-Do
+						 * Case 6,8 needs special handling to convert the
+						 * given numeric to date format(YYYY-MM-DD/YY-MM-DD)
+						 */
+						case 4:
+						case 6:
+						case 8:
+							break;
+						/*
+						 * If the numeric is of length {3,5,7, >8} then an error should be thrown*/
+						default:
+							gen_func_time_in_error();
+							break;
+					}
+			}
+		}
+		switch(nf)
+		{
+			case 1:
+				/*
+				 * To-Do
+				 * If the input is of format `Mon{/.}yyyy{/.}dd` or `Mon{/.}dd{/.}yyyy
+				 * shouldn't be supported.
+				 */
+				/*
+				 * If only date is specified add an default time of
+				 * 0:0:0
+				 */
+				if(ftype[0] == DTK_DATE)
+				{
+					ftype[1] = DTK_TIME;
+					field[1] = "0:0:0";
+					nf = nf+1;
+ 				}
+				break;
+			case 3:
+				/*
+				 * If the given input is of format `yyyy-mm-ddThh:mm:ss`
+				 * then conver it to `yyyy-mm-dd hh:mm:ss` by ignoring 'T'
+				 */
+				if(ftype[1] == DTK_STRING && pg_strcasecmp(field[1], "t") == 0 && ftype[2] == DTK_TIME)
```

Comment:
why small "t"?

Reply:
We are comparing with only small "t" because after parsing even if we given "T" will be converted to "t".